devito/ir/equations/algorithms.py

```diff
@@ -64,3 +65,52 @@ def handle_indexed(indexed):
     ordering = PartialOrderTuple(extra, relations=(relations | implicit_relations))
 
     return ordering
+
+
+def lower_exprs(expressions, **kwargs):
+    """
+    Expression lowering and indexification:
+
+        * Indexify functions
+        * Align indexeds with the computational domain
+        * Consider user-applied subs
+
+    # E.g., f(x - 2*h_x, y) -> f[xi + 2, yi + 4]  (assuming halo_size=4)
+    """
+
+    processed = []
+    for expr in expressions:
+        try:
```

Comment:
what's going on here? this is ugly

Reply:
would rather avoid try/except too but even if necessary, the `if` is completely unncecessary 
```
try: 
    dimension_map = expr.subdomain.dimension_map 
except AttributeError:
   dimension_map ={}
```
does the same.

Reply:
> Some condition expressions do not have a subdomain 

such as which ones?

Reply:
> such as which ones?

this is being discussed on Slack -- about overriding `__lt__` etc to return our own `Lt` etc, which do have a subdomain attached

Reply:
if we don't wanna fix it now, perhaps leave a comment

```
try:
    ...
except AttributeError:
    # Some Relationals may be pure SymPy objects, thus lacking the subdomain
    dimension_map = {}
```